From 717ee459a4d08b2b68b1ad22a62403a567ca3b0d Mon Sep 17 00:00:00 2001
From: Reinette Chatre <reinette.chatre@intel.com>
Date: Sat, 25 Feb 2012 10:54:11 +0200
Subject: [PATCH 46/58] dbus: return NULL data for peer IEs

The org.freedesktop.DBus.Properties interface is now supported for peer
properties. Ensure that GetAll will work by having the IEs property
function return NULL data instead of a failure until it is implemented.

Signed-hostap: Reinette Chatre <reinette.chatre@intel.com>
intended-for: hostap-1
(cherry picked from commit 93d1749fe09522892779151f0e22adb3e52efa20)
---
 wpa_supplicant/dbus/dbus_new_handlers_p2p.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/wpa_supplicant/dbus/dbus_new_handlers_p2p.c
+++ b/wpa_supplicant/dbus/dbus_new_handlers_p2p.c
@@ -1465,10 +1465,12 @@ dbus_bool_t wpas_dbus_getter_p2p_peer_ve
 dbus_bool_t wpas_dbus_getter_p2p_peer_ies(DBusMessageIter *iter,
 					  DBusError *error, void *user_data)
 {
+	dbus_bool_t success;
 	/* struct peer_handler_args *peer_args = user_data; */
 
-	dbus_set_error_const(error, DBUS_ERROR_FAILED, "not implemented");
-	return FALSE;
+	success = wpas_dbus_simple_array_property_getter(iter, DBUS_TYPE_BYTE,
+							 NULL, 0, error);
+	return success;
 }
 
 
